Zebble/Android/Controls/Scrolling/AutoScrollUtility.cs

```diff
@@ -0,0 +1,179 @@
+
+
+        private bool IsPartOfViewVisible(ScrollView scrollViewParent, View view)
+        {
+            Rect scrollBounds = new Rect();
+            scrollViewParent.GetHitRect(scrollBounds);
+            // Any portion of the imageView, even a single pixel, is within the visible window
+            return view.GetLocalVisibleRect(scrollBounds);
+        }
+
+        /**
+         * Used to scroll to the given view.
+         *
+         * @param scrollViewParent Parent ScrollView
+         * @param view View to which we need to scroll.
+         */
+        private void ScrollToView(ScrollView scrollViewParent)
+        {
+            try
+            {
+                // Any portion of the imageView, even a single pixel, is within the visible window
+                if (IsViewFullyVisible(scrollViewParent))
+                {
+                    return;
+                }
+
+                Point childOffset = new Point();
+                GetDeepChildOffset(_parentScrollView, _view.Parent, _view, childOffset);
+                // Scroll to child.
+                scrollViewParent.SmoothScrollTo(0, childOffset.Y);
+            }
+            catch (ObjectDisposedException)
+            {
+                Dispose();
+            }
+        }
+
+        /**
+         * Used to get deep child offset.
+         * <p/>
+         * 1. We need to scroll to child in scrollview, but the child may not the direct child to scrollview.
+         * 2. So to get correct child position to scroll, we need to iterate through all of its parent views till the main parent.
+         *
+         * @param mainParent        Main Top parent.
+         * @param parent            Parent.
+         * @param child             Child.
+         * @param accumulatedOffset Accumulated Offset.
+         */
+        private void GetDeepChildOffset(ViewGroup mainParent, IViewParent parent, View child, Point accumulatedOffset)
+        {
+            ViewGroup parentGroup = (ViewGroup)parent;
+            accumulatedOffset.X += child.Left;
+            accumulatedOffset.Y += child.Top;
+            if (parentGroup.Equals(mainParent))
+            {
+                return;
+            }
+            GetDeepChildOffset(mainParent, parentGroup.Parent, parentGroup, accumulatedOffset);
+        }
+
+        /**
+         * Used to get real child offset even its not visible.
+         * <p/>
+         * 1. We need to scroll to child in scrollview, but the child may not the direct child to scrollview.
+         * 2. So to get correct child position to scroll, we need to iterate through all of its parent views till the main parent.
+         *
+         * @param mainParent        Main Top parent.
+         * @param parent            Parent.
+         * @param child             Child.
+         * @param accumulatedOffset Accumulated Offset.
+         */
+        private void GetDeepChildRealPosition(ViewGroup mainParent, IViewParent parent, View child, Point accumulatedOffset)
```

Comment:
@Ali-YousefiTelori Instead of passing "accumulatedOffset" as a parameter change the type of the method to Point and return it in the body of the method.